src/sagemaker/model.py

```diff
@@ -242,10 +394,90 @@ def prepare_container_def(
         Returns:
             dict: A container definition object usable with the CreateModel API.
         """
+        deploy_key_prefix = fw_utils.model_code_key_prefix(
+            self.key_prefix, self.name, self.image_uri
+        )
+        deploy_env = copy.deepcopy(self.env)
+        if self.source_dir or self.dependencies or self.entry_point or self.git_config:
+            if self.key_prefix or self.git_config:
+                self._upload_code(deploy_key_prefix, repack=False)
+            elif self.source_dir and self.entry_point:
+                self._upload_code(deploy_key_prefix, repack=True)
+            else:
+                self._upload_code(deploy_key_prefix, repack=False)
+            deploy_env.update(self._script_mode_env_vars())
         return sagemaker.container_def(
```

Comment:
Can we concise this as:
```
if self.source_dir and self.entry_point:
	self._upload_code(deploy_key_prefix, repack=True)
else:
	self._upload_code(deploy_key_prefix, repack=False)
```

and with that you can even use ternary to condense it into a single line further